akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,22 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case sel @ ActorSelectionMessage(msg, elements, _) ⇒ {
+      elements.head match {
+        case SelectChildName(name) ⇒
+          val child = getChild(name)
+          if (elements.tail.isEmpty) {
+            child ! msg
+          } else {
+            child ! sel.copy(elements = elements.tail)
+          }
+        case _ ⇒ // .. and * are unsupported
```

Comment:
Logging?

-- 
Cheers,
√
On 8 Feb 2015 13:27, "Roland Kuhn" <notifications@github.com> wrote:

> In akka-actor/src/main/scala/akka/actor/ActorRef.scala
> <https://github.com/akka/akka/pull/16690#discussion_r24298199>:
>
> > @@ -599,6 +599,22 @@ private[akka] class VirtualPathContainer(
> >
> >    private val children = new ConcurrentHashMap[String, InternalActorRef]
> >
> > +  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
> > +    case sel @ ActorSelectionMessage(msg, elements, _) ⇒ {
> > +      elements.head match {
> > +        case SelectChildName(name) ⇒
> > +          val child = getChild(name)
> > +          if (elements.tail.isEmpty) {
> > +            child ! msg
> > +          } else {
> > +            child ! sel.copy(elements = elements.tail)
> > +          }
> > +        case _ ⇒ // .. and * are unsupported
>
> Since tell cannot throw the only thing we can do is to send to
> deadLetters (which I proposed in this comment thread already), unless I
> overlooked something.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/16690/files#r24298199>.
>